Testfailure : test_prune_split[hourly-3-expected_ids4] assertionerror

## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG

## System information. For client/server mode post info for both machines.

#### Your borg version (borg -V).
borg 1.4-maint branch as of 2025-03-07

#### Operating system (distribution) and version.
MacOS 15.1.1
Time Zone: Asia/Kolkata
python version 3.13.1

## Describe the problem you're observing.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
(1.4-maint branch)Yes, `tox` / `pytest -vv`

#### Include any warning/errors/backtraces from the system logs
```
FAILED src/borg/testsuite/helpers.py::test_prune_split[hourly-3-expected_ids4] - assert {13: 2017-10-01T10:00:05+00:00, 7: 2017-03-03T10:00:00+00:00, 8: 2017-03-04T10:00:00+00:00} == {13: 2017-10-01T10:00:05+00:00, 10: 2017-10-01T09:55:00+00:00, 8: 2017-03-04T10:00:00+00:00}
  
  Extra items in the left set:
  7: 2017-03-03T10:00:00+00:00
  Extra items in the right set:
  10: 2017-10-01T09:55:00+00:00
  
  Full diff:
    {
        13: 2017-10-01T10:00:05+00:00,
  -     10: 2017-10-01T09:55:00+00:00,
        8: 2017-03-04T10:00:00+00:00,
  +     7: 2017-03-03T10:00:00+00:00,
    }
```
```
  py39-none: FAIL code 1 (645.12=setup[57.10]+cmd[588.02] seconds)
  py39-fuse2: FAIL code 1 (670.43=setup[49.75]+cmd[620.67] seconds)
  py39-fuse3: FAIL code 1 (5.20 seconds)
  py310-none: FAIL code 1 (591.63=setup[58.04]+cmd[533.59] seconds)
  py310-fuse2: FAIL code 1 (712.46=setup[108.09]+cmd[604.37] seconds)
  py310-fuse3: FAIL code 1 (3.49 seconds)
  py311-none: FAIL code 1 (588.10=setup[39.14]+cmd[548.96] seconds)
  py311-fuse2: FAIL code 1 (629.90=setup[45.12]+cmd[584.78] seconds)
  py311-fuse3: FAIL code 1 (2.95 seconds)
  py312-none: SKIP (0.02 seconds)
  py312-fuse2: SKIP (0.01 seconds)
  py312-fuse3: SKIP (0.01 seconds)
  py313-none: FAIL code 1 (661.26=setup[61.17]+cmd[600.08] seconds)
  py313-fuse2: FAIL code 1 (756.85=setup[115.07]+cmd[641.79] seconds)
  py313-fuse3: FAIL code 1 (3.31 seconds)
  evaluation failed :( (5270.84 seconds)
```

--- a/src/borg/testsuite/helpers.py
+++ b/src/borg/testsuite/helpers.py
@@ -2,6 +2,7 @@ import hashlib
 import os
 import shutil
 import sys
+import time
 from argparse import ArgumentTypeError
 from datetime import datetime, timezone, timedelta
 from io import StringIO, BytesIO
@@ -392,6 +393,12 @@ class MockArchive:
     def __repr__(self):
         return f"{self.id}: {self.ts.isoformat()}"
 
+@pytest.fixture
+def set_prune_test_timezone(monkeypatch):
+    """Run prune tests in a specific full hour timezone"""
+    monkeypatch.setenv("TZ", "Etc/GMT-1")
+    if hasattr(time, "tzset"):  # Only call tzset() on Unix
+        time.tzset()
 
 @pytest.mark.parametrize(
     "rule,num_to_keep,expected_ids", [
@@ -399,13 +406,15 @@ class MockArchive:
         ("monthly", 3, (13, 8, 4)),
         ("weekly", 2, (13, 8)),
         ("daily", 3, (13, 8, 7)),
+        # the "hourly" test case fails if the timezone is not a full-hour offset from UTC. 
+        # therefore, timezone is set to Etc/GMT-1 in set_prune_test_timezone.
         ("hourly", 3, (13, 10, 8)),
         ("minutely", 3, (13, 10, 9)),
         ("secondly", 4, (13, 12, 11, 10)),
         ("daily", 0, []),
     ]
 )
-def test_prune_split(rule, num_to_keep, expected_ids):
+def test_prune_split(rule, num_to_keep, expected_ids, set_prune_test_timezone):
     def subset(lst, ids):
         return {i for i in lst if i.id in ids}
 

